.cbmc-batch/stubs/evp_pkey_is_valid_stub.c

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use
+ * this file except in compliance with the License. A copy of the License is
+ * located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <make_common_data_structures.h>
+#include <openssl/evp.h>
+
+/* Abstraction of the EVP_PKEY struct */
+struct evp_pkey_st {
+    int references;
+    EC_KEY *ec_key;
+};
+
+/**
+ * Helper function for CBMC proofs: checks if EVP_PKEY is valid.
+ * Use this stub when we are *certain* there is no ec_key associated with the key.
+ */
+bool evp_pkey_is_valid(EVP_PKEY *pkey) {
+    return pkey && (pkey->references > 0) && (pkey->ec_key == NULL);
```

Comment:
`ec_key == null` is the only valid value?

Reply:
Not in general, no and the more generic evp_pkey_is_valid function reflects this. I introduce this stub to improve coverage in the case where we are *certain* that ec_key == null for the function being addressed in the CBMC proof. This greatly improves coverage as otherwise a string of unreached functions occurs. I have renamed both stubs in this PR to make this clear.